### PR TITLE
Aws - Kinesis Video Streams, Delete action, kms-key filter

### PR DESCRIPTION
**Pull request by Lucas-Irvine**
Closes https://github.com/cloud-custodian/cloud-custodian/issues/6489

Add support for the `kinesis-video` resource. Also add a `delete` action and `kms-key` filter.

### CODE REVIEW

File: c7n/resources/kinesis.py [hunk 1]
```diff
@@ -270,3 +270,85 @@ def process(self, resources):
...
+            actions:
+              - type: delete
+    """
+
+    schema = type_schema('delete')
+    permissions = ("kinesisvideo:DeleteStream",)
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('kinesisvideo')
+        not_active = [r['StreamName'] for r in resources
```
Comment (kapilt) [inline c7n/resources/kinesis.py:322]: this is provided by a builtin in element.py filter_resources

File: c7n/resources/kinesis.py [hunk 2]
```diff
@@ -270,3 +270,85 @@ def process(self, resources):
...
+                      if r['Status'] != 'ACTIVE']
+        self.log.warning(
+            "The following streams cannot be deleted (wrong state): %s" % (
+                ", ".join(not_active)))
+        for r in resources:
+            if not r['Status'] == 'ACTIVE':
```
Comment (kapilt) [inline c7n/resources/kinesis.py:328]: why check twice

File: c7n/resources/kinesis.py [hunk 3]
```diff
@@ -270,3 +270,85 @@ def process(self, resources):
...
+                continue
+            client.delete_stream(
```
Comment (kapilt) [inline c7n/resources/kinesis.py:330]: handle resource not found

**Comment by Lucas-Irvine** [comment]
Thanks for the review @kapilt! Changes should be addressed.

**Review by kapilt — APPROVED** [review]
lgtm, thanks

**Pull request merged**